presto-spi/src/main/java/com/facebook/presto/spi/type/MapType.java

```diff
@@ -245,4 +246,24 @@ public Block createBlockFromKeyValue(boolean[] mapIsNull, int[] offsets, Block k
                 keyNativeHashCode,
                 keyBlockHashCode);
     }
+
+    /**
+     * Create a map block directly without per element validations.
+     *
+     * Internal use by com.facebook.presto.spi.Block only.
+     */
+    public static Block createMapBlockInternal(
+            TypeManager typeManager,
+            Type keyType,
+            int positionCount,
+            boolean[] mapIsNull,
+            int[] offsets,
+            Block keyBlock,
+            Block valueBlock,
+            int[] hashTables)
+    {
+        // TypeManager caches types. Therefore, it is important that we go through it instead of coming up with the MethodHandles directly.
+        MapType mapType = (MapType) typeManager.getType(new TypeSignature(StandardTypes.MAP, TypeSignatureParameter.of(keyType.getTypeSignature()), TypeSignatureParameter.of(BIGINT.getTypeSignature())));
+        return MapBlock.createMapBlockInternal(0, positionCount, mapIsNull, offsets, keyBlock, valueBlock, hashTables, keyType, mapType.keyBlockNativeEquals, mapType.keyNativeHashCode);
```

Comment:
consider changing the signature of `MapBlock.createMapBlockInternal` to match closer to the signature of `MapType.createMapBlockInternal`